ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -72,6 +74,22 @@ let ntlm_auth uname passwd : (unit, exn) result =
     Ok ()
   with _ -> Error (auth_ex uname)
 
+let kdc_of_domain ~domain ~msg =
+  let hd msg = function
+    | [] ->
+        error "%s" msg ;
+        raise (Auth_service_error (E_LOOKUP, msg))
+    | h :: _ ->
+        h
+  in
+  Helpers.call_script ~log_output:On_failure net_cmd
+    ["lookup"; "kdc"; domain; "-d"; debug_level ()]
+  (* Result like 10.71.212.25:88\n10.62.1.25:88\n*)
+  |> String.split_on_char '\n'
+  |> hd "lookup kdc return invalid result"
+  |> String.split_on_char ':'
+  |> hd "kdc has invalid address"
+
```

Comment:
I wonder if these IP addresses come in any particular order (e.g. sorted by round-trip time). I couldn't find any documentation on this

Reply:
Good question!
I did several times, and just found they comes randomly. 

I am currently working on another ticket to improve the ldap performance, basically, will query the latest kdc and persist in xapi database. I will raise PR when it is ready, 
For this PR, just keep it for now.